store/supabase.js

```diff
@@ -0,0 +1,133 @@
+// state
+export const state = () => ({
+  topics: [],
+  members: [],
+  edges: [],
+  nodes: [],
+});
+
+export const actions = {
+  async fetchData({ commit }) {
+    // hit supabase to get topics, members and mappings
+    const { data: topics, error: topicError } = await this.$supabase
+      .from("topics")
+      .select("*");
+    const { data: members, error: memberError } = await this.$supabase
+      .from("members")
+      .select("*");
+    const { data: mapping, error: mappingError } = await this.$supabase
+      .from("members_topics")
+      .select("*");
+
+    if (!topicError) {
+      commit("setTopics", topics);
+    }
+    if (!memberError) {
+      commit("setMembers", members);
+    }
+
+    if (!topicError && !memberError && !mappingError) {
+      const { edges, nodes } = createNodes(members, topics, mapping);
+      commit("setEdges", edges);
+      commit("setNodes", nodes);
+    }
+  },
+};
+
+export const mutations = {
+  setTopics(state, topics) {
+    state.topics = topics;
+  },
+  setMembers(state, members) {
+    state.members = members;
+  },
+  setEdges(state, edges) {
+    state.edges = edges;
+  },
+  setNodes(state, nodes) {
+    state.nodes = nodes;
+  },
+};
+
+function createNodes(members, topics, mappings) {
+  let nodes = [];
+  let memberNames = {};
+  for (const member of members) {
+    nodes.push({
+      _cssClass: "Member",
```

Comment:
the createNodes function prepares the graph data structure specifically as expected by the `vue-d3-network` lib, and thus sets a lot of display attributes like `_cssClass` and `_svgAttrs`. 
I think this should not be the responsibility of the store and therefore this function could better be defined in an external utility file.